new-frontiers-14/frontier-station-14
# Add Hovertaxibot

## About the PR



Adds a Hovertaxibot (pending a better name), essentially a taxibot that's space-capable.  Available as a ghost role, it has a built in radar system (thanks, Leander), a horn, and moves as fast as a hoverbike.  Like the taxibot, the rider _cannot control it_, only the person taking the ghost role can.

I created a new VehicleHornComponent for things that should have horns but are not vehicles (e.g. taxibots), because the rest of the sound emission is annoying.

Robots now use the pAI speech sounds (they should have _something_), and a few missing localization strings were added for taxibots (and some lingering ones removed).

## Why / Balance


The Taxibot isn't very good for Frontier Station, it's mostly space.  Enter the Hovertaxibot.

With regards to balance, there isn't much the ghost role can do outside of honk and fly around.  A user may climb onboard a hovertaxibot and be flown immediately out of space and have them disconnect, but at that point, it seems like a user issue.  Not terribly griefable, and the horn has the same cooldown as a normal vehicle's (same action).

I didn't like the tag added for the thruster board, but I figured it would make more sense to add some minor complexity to the construction graph.

## How to test


1. Construct a hoverbike, follow the in-game construction steps.
2. Hop onboard.
3. Connect in a separate client, join as an observer, join as the hovertaxibot.
4. Note that no localization strings are missing from the description/name of the hovertaxibot, and you should be joining as a non-antag.
5. Fly around, you should move like a hoverbike.
6. Bring up your radar screen, it should work.
7. Honk your horn, it should sound.  Beep beep.
8. Fly into space, you should still have control.
9. Say something, you should make a noise like a pAI.
10. Try to interact with the environment, you should not be able to.
11. Run `/ghost`, your horn action should be gone.
12. Spawn a taxibot, join as a taxibot.
13. Note that no localization strings are missing from the description/name of the taxibot, and you should be joining as a non-antag.
14. Honk your horn, it should sound.  Beep beep.
15. Say something, you should make a noise like a pAI.
16. Run `/ghost`, your horn action should be gone.
17. Spawn a mimebot, join as the mimebot.
18. Say something (blasphemy), you should not make a sound.

## Media


https://github.com/user-attachments/assets/390d3f36-280b-43de-8afb-ea91137864df


## Breaking changes


**Changelog**



:cl:
- add: The Hovertaxibot, a spacefaring taxi robot with a ghost role, is now craftable.
